### DISCUSSION TITLE
AutoComplete for Entry/Editor controls

### DISCUSSION THREAD

**Discussion by ThysBrits** [CommunityToolkit/Maui#1824] [New Feature Discussions]
It would be great if the Entry and/or Editor controls could have the ability to have AutoComplete. I believe there are paid-for controls (e.g. SyncFusion) that have this, but this seems like it should be something that is natively supported.

**Comment by bijington** [comment]
Do you have any thoughts on how this might work? Are you proposing new controls or some level of extension to the existing MAUI controls?

**Reply by prabhav-mehra** [reply]
I would think as a new component. The main aim is, 
- you can populate the autocomplete with sort of collection of items.
- Assign which property to look for (if you have assigned a list of objects and not just string)
- And when you type in the entry, it filters and shows the matched property out of the collection.
- Maybe a system where it auto-fills the entry as well or allow the user to choose it explicitly.
- Configurable UI of dowpdown-ish ui for list of items, ui for selected item, allow user to specify select one or select multiple?

These are some high level requirements

**Reply by prabhav-mehra** [reply]
Maybe the implementation extends the Entry base class and builds upon it. The reason I suggested a different control is to avoid a simple entry to have such complex implementation

**Reply by bijington** [reply]
@s3743761 would you be willing to contribute to the creation of such a control?

**Reply by prabhav-mehra** [reply]
I haven't contributed here yet but I'd be happy to. I'll create an issue with some requirements and have a look into it.

**Reply by pictos** [reply]
@s3743761 before creating an issue, please create a discussion with the API design that you've in mind. You can also link controls (from .net or android/ios) that we can use as reference

**Comment by CliffAgius** [comment]
Just a poke to see if @s3743761 you are going to add a API design?

Otherwise I will close this discussion as it's become stale.

**Comment by mos379** [comment]
@prabhav-mehra any chance you would follow up on this?

**Comment by stephenquan** [comment]
Perhaps a new component that wraps entry, e.g. `public class AutoComplete : ContentView` but it would implement IEntry so that users could swap out `Entry` for `AutoComplete`.

Otherwise, a custom Popup, say `public class AutoCompletePopup : Popup`, but it means that the user needs to wire it up in C# code-behind.
